projects/hipblaslt/library/src/amd_detail/rocblaslt/src/rocroller_host.cpp

```diff
@@ -748,6 +748,12 @@ int chooseStreamKGridSize(std::shared_ptr<GemmKernel>        gemm,
     else
         dataType = rocroller_type_to_analytical_type(gemm->params->kernelType.typeA);
 
+    auto reduction_type = TensileLite::analytical::streamk::select_streamk_reduction(prob.m, prob.n, prob.k, prob.batch_count,
+        gemm->params->workgroupTile.m, gemm->params->workgroupTile.n, gemm->params->workgroupTile.k, analaytical_hardware, 6);
```

Comment:
Can we do something like this?

```suggestion
    const int cus_per_xcd = analytical_hardware.N_CU / analytical_hardware.NUM_XCD;
    const int wgm= std::ceil(std::sqrt(cus_per_xcd));
    auto reduction_type = TensileLite::analytical::streamk::select_streamk_reduction(prob.m, prob.n, prob.k, prob.batch_count,
        gemm->params->workgroupTile.m, gemm->params->workgroupTile.n, gemm->params->workgroupTile.k, analaytical_hardware, wgm);
```

Reply:
The 6 here is dynamic grid mode, not WGM. Probably should have a named constant for clarity

Reply:
Pushed a changed adding a named constant